### ISSUE TITLE
Using Concatenate&ParamSpec leads to skipped parameter in popup-window.

### ISSUE THREAD

**Issue by GugelRobin** [microsoft/pylance-release#3275]
<h3> Environment data </h3>

-   Language Server version: 2022.8.50
-   OS and version: darwin arm64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.8.13
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
P = ParamSpec("P")
def tester(c: Callable[Concatenate[int, str, P], None]) -> Callable[Concatenate[int, P], None]:
    pass

@tester
def foo(a: int, b: str, c: str, d: str):
    pass
```

Using python 3.8 and typing_extensions package.
The first parameter `P` captures isn't  shown in the popup-window when calling `foo`.

Note that this does not happen when using just `P` as the return value.

<h3> Expected behavior </h3>

Parameter 'c' should be coloured blue.

<h3> Actual behavior </h3>

Parameter 'd' is assumed to be the next parameter.
![image](https://user-images.githubusercontent.com/31984132/188190852-de898c6d-5655-4b13-b4b2-f9a54373b5c0.png)

**Comment by GugelRobin** [comment]
Also:
* If each argument is documented the correct argument documentation is shown while the blue 'you're currently at that argument'-marker is still wrong.

* When naming the parameter (e.g. `foo(3, c="h...`) the behaviour still holds that one parameter to the right is blue - even with more parameters.

**Comment by erictraut** [comment]
Yes, this looks like a bug in the "signature help provider", which is the part of pylance responsible for providing the signature help displayed in this popup window.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2022.9.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022921-14-September-2022-PreRelease)
